src/server/routes/backend-wallet/withdraw.ts

```diff
@@ -62,45 +70,69 @@ export async function withdraw(fastify: FastifyInstance) {
     },
     handler: async (request, reply) => {
       const { chain: chainQuery } = request.params;
-      const { toAddress } = request.body;
-      const {
-        "x-backend-wallet-address": walletAddress,
-        "x-idempotency-key": idempotencyKey,
-      } = request.headers as Static<typeof walletHeaderSchema>;
+      const { toAddress, txOverrides } = request.body;
+      const { "x-backend-wallet-address": walletAddress } =
+        request.headers as Static<typeof walletHeaderSchema>;
 
       const chainId = await getChainIdFromChain(chainQuery);
       const chain = await getChain(chainId);
-      const from = walletAddress as Address;
+      const from = getChecksumAddress(walletAddress);
+
+      // Populate a transfer transaction with 2x gas.
+      const populatedTransaction = await toSerializableTransaction({
+        from,
+        transaction: {
+          to: toAddress,
+          chain,
+          client: thirdwebClient,
+          data: "0x",
+          // Dummy value, replaced below.
+          value: 1n,
+          gas: maybeBigInt(txOverrides?.gas),
+          maxFeePerGas: maybeBigInt(txOverrides?.maxFeePerGas),
+          maxPriorityFeePerGas: maybeBigInt(txOverrides?.maxPriorityFeePerGas),
+        },
+      });
+
+      // Compute the maximum amount to withdraw taking into account gas fees.
+      const value = await getWithdrawValue(from, populatedTransaction);
+      populatedTransaction.value = value;
 
       const account = await getAccount({ chainId, from });
-      const value = await getWithdrawValue({ chainId, from });
+      let transactionHash: Hex | undefined;
+      try {
+        const res = await account.sendTransaction(populatedTransaction);
+        transactionHash = res.transactionHash;
+      } catch (e) {
+        logger({
+          level: "warn",
```

Comment:
It may not be a _system_ error. For example if someone calls this on a wallet that just has too little dust to make any transfer, this will be attempted and fail.